src/FakeItEasy/Creation/ProxyOptions.cs

```diff
@@ -25,7 +26,8 @@ internal class ProxyOptions : IProxyOptions
 
         public IEnumerable<Expression<Func<Attribute>>> Attributes => this.attributes;
 
-        public string Name { get; set; }
+        [DisallowNull]

```

Comment:
Is there any benefit to adding this attribute here? We never explicitly set it to null, but we could; it's internal anyway. Also we don't check if the assigned value is null, so the attribute basically a lie... Unless it's necessary to quell a warning, I'd be inclined to remove it.